tests/Oryx.BuildImage.Tests/ExecCommandTest.cs

```diff
@@ -0,0 +1,92 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using Microsoft.Oryx.Tests.Common;
+using Xunit;
+using Xunit.Abstractions;
+using Microsoft.Oryx.BuildScriptGenerator.Node;
+using Microsoft.Oryx.Common;
+using Microsoft.Oryx.BuildScriptGenerator.Php;
+
+namespace Microsoft.Oryx.BuildImage.Tests
+{
+    public class ExecCommandTest : SampleAppsTestBase
+    {
+        public ExecCommandTest(ITestOutputHelper output) : base(output) { }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData("/bin/dash")]
+        public void Exec_NodeLtsVersion(string altBashPath)
+        {
+            // Arrange
+            var appPath = "/tmp";
+            var cmd = "node --version";
+
+            var scriptBuilder = new ShellScriptBuilder()
+                .CreateFile($"{appPath}/{NodeConstants.PackageJsonFileName}", "{}");
+
+            var expectedBashPath = FilePaths.Bash;
+            if (!string.IsNullOrEmpty(altBashPath))
+            {
+                scriptBuilder.SetEnvironmentVariable("BASH", altBashPath);
+                expectedBashPath = altBashPath;
+            }
+
+            var script = scriptBuilder
+                .AddCommand($"oryx exec --debug --src {appPath} '{cmd}'") // '--debug' prints the benv command
+                .ToString();
+
+            // Act
+            var result = _dockerCli.Run(Settings.BuildImageName, "/bin/bash", "-c", script);
+
+            // Assert
+            RunAsserts(
+                () =>
+                {
+                    var nodeArg = $"node={NodeConstants.NodeLtsVersion}";
+                    Assert.Contains("#!" + expectedBashPath, result.StdOut);
+                    Assert.Contains(nodeArg, result.StdOut);
+                    Assert.True(result.IsSuccess);
+                    // Actual output from `node --version` starts with a 'v'
+                    Assert.Contains($"v{NodeConstants.NodeLtsVersion}", result.StdOut);
+                },
+                result.GetDebugInfo());
+        }
+
+        [Fact]
+        public void Exec_NodeVersionAndPhpVersion()
```

Comment:
Rename the test so that it indicates what the pivot here actually is...CanExecuteMultipleCommands may be?